File: java/pattern/src/observer/ObserverA.java
package observer;

public class ObserverA implements Observer {
    // myState需要跟目标对象的state值保持一致
    private int myState;

    @Override
    public void update(Subject s) {
        myState = ((ConcreteSubject) s).getState();
    }

    public int getMyState() {
        return myState;
    }

    public void setMyState(int myState) {
        this.myState = myState;
    }
}

File: php/observer/ObserverA.php
<?php


class ObserverA implements Observer
{
    // myState需要跟目标对象的state值保持一致
    private $myState;

    public function update($state)
    {
        $this->myState = $state->getState();
    }

    public function getMyState()
    {
        return $this->myState;
    }

    public function setMyState($state)
    {
        $this->myState = $state;
    }
}

